launcher/src/main/java/org/apache/spark/launcher/AbstractCommandBuilder.java

```diff
@@ -89,7 +89,7 @@
    */
   List<String> buildJavaCommand(String extraClassPath) throws IOException {
     List<String> cmd = new ArrayList<>();
-    String javaCommand = System.getenv("JAVA_CMD");
+    String javaCommand = System.getenv("JAVA_COMMAND");
```

Comment:
This is a fix for a previous patch I made. It looks like `JAVA_CMD` will conflict to some of env var Spark made and thus break it.